Restart core services based on changes to Java

kiiranh opened the pull request:
This PR fixes issue #136. The following services now restart whenever JAVA_HOME is changed:
* Namenode (master, no_HA, standby)
* Journalnode
* Hive
* Flume
* Hbase (master, region server)
* Httpfs
* Zookeeper

kiiranh commented:
This has been tested with HDP 2.0 on vbox vm cluster and then rebased on top of the hdp2.2 changes.
Will be squashed into one commit after review.

cbaenziger commented on cookbooks/bcpc-hadoop/recipes/copylog.rb:
```diff
@@ -38,7 +38,10 @@
 # Currently we use the default flume-env values
```
Can probably nix this comment now...

cbaenziger commented:
Kiran, I do not think the merge went quite right; there should already be a [zookeeper-env.sh](https://github.com/bloomberg/chef-bach/commit/a1b664ccf3fac73919fa9885cf65cbc86b56f6f7#diff-23485cd9a22ee669fa8563a923003bc2) introduced by HDP2.2.

kiiranh commented:
Clay, the hdp2.2 changes add new files with the prefix 'hdp' for zookeeper files. I've renamed them to have prefix 'zk'. Also updated the code which referred to these templates.

kiiranh commented:
This changes have been tested on a vbox vm cluster. Ready for review/merge.

cbaenziger commented:
Test on a hardware cluster too
